src/cli/backup/exchange_e2e_test.go

```diff
@@ -35,6 +35,184 @@ var (
 	events   = path.EventsCategory
 )
 
+// ---------------------------------------------------------------------------
+// tests with azure flags in exchange create
+// ---------------------------------------------------------------------------
+
+type ExchangeCMDWithFlagsE2ESuite struct {
+	tester.Suite
+	acct       account.Account
+	st         storage.Storage
+	vpr        *viper.Viper
+	cfgFP      string
+	repo       repository.Repository
+	m365UserID string
+	recorder   strings.Builder
+}
+
+func TestBackupExchangeInvalidFlagValueSuite(t *testing.T) {
+	suite.Run(t, &ExchangeCMDWithFlagsE2ESuite{Suite: tester.NewE2ESuite(
+		t,
+		[][]string{tester.AWSStorageCredEnvs, tester.M365AcctCredEnvs},
+	)})
+}
+
+func (suite *ExchangeCMDWithFlagsE2ESuite) SetupSuite() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	acct, st, repo, vpr, recorder, cfgFilePath := prepM365Test(t, ctx)
+
+	suite.acct = acct
+	suite.st = st
+	suite.repo = repo
+	suite.vpr = vpr
+	suite.recorder = recorder
+	suite.cfgFP = cfgFilePath
+	suite.m365UserID = tester.M365UserID(t)
+}
+
+func (suite *ExchangeCMDWithFlagsE2ESuite) TestExchangeBackupInvalidAzureClientIDCmd_empty() {
+	t := suite.T()
+	ctx, flush := tester.NewContext(t)
+
+	defer flush()
+
+	suite.recorder.Reset()
+
+	cmd := tester.StubRootCmd(
+		"backup", "create", "exchange",
+		"--user", suite.m365UserID,
+		"--azure-client-id", "invalid-value",
+	)
+	cli.BuildCommandTree(cmd)
+
+	cmd.SetErr(&suite.recorder)
+
+	ctx = print.SetRootCmd(ctx, cmd)
+
+	// run the command
+	err := cmd.ExecuteContext(ctx)
+	require.Error(t, err, clues.ToCore(err))
+}
+
+func (suite *NoBackupExchangeE2ESuite) TestExchangeBackupValueFromConfigCmd_empty() {
+	t := suite.T()
+	ctx, flush := tester.NewContext(t)
+	ctx = config.SetViper(ctx, suite.vpr)
+
+	defer flush()
+
+	suite.recorder.Reset()
+
+	cmd := tester.StubRootCmd(
+		"backup", "create", "exchange",
+		"--user", suite.m365UserID,
+		"--config-file", suite.cfgFP)
+	cli.BuildCommandTree(cmd)
+
+	cmd.SetErr(&suite.recorder)
+
+	ctx = print.SetRootCmd(ctx, cmd)
+
+	// run the command
+	err := cmd.ExecuteContext(ctx)
+	require.NoError(t, err, clues.ToCore(err))
+
+	result := suite.recorder.String()
+	t.Log("backup results", result)
+
+	// as an offhand check: the result should contain the m365 user id
+	assert.Contains(t, result, suite.m365UserID)
+}
+
+func (suite *NoBackupExchangeE2ESuite) TestExchangeBackupValueFromEnvCmd_empty() {
```

Comment:
These are all just variations on what values we accept in order to run a backup, right?  That'll end up running one complete, end-to-end backup for each test, which isn't wanted.  Lets avoid additional e2e tests that aren't focused on error cases or else we'll bloat ci times pretty quickly.  These sorts of conditions are ideally testable with a mocked interface so that we can evaluate all the logic in the cli layer without making any calls to data producers or consumers.

Reply:
Can you please point me to one such test case?